go/service/usersearch.go

```diff
@@ -197,18 +202,51 @@ func (h *UserSearchHandler) UserSearch(ctx context.Context, arg keybase1.UserSea
 		return res, nil
 	}
 
+	if arg.Service != "keybase" && arg.Service != "" {
+		// If this is a social search, we just return API results.
+		return doSearchRequest(mctx, arg)
+	}
+
 	res, err = doSearchRequest(mctx, arg)
+	if err != nil {
+		mctx.Warning("Failed to do an API search for %q: %s", arg.Service, err)
+	}
+
 	if arg.IncludeContacts {
 		contactsRes, err := contactSearch(mctx, arg)
 		if err != nil {
-			return nil, err
-		}
-		if len(contactsRes) > 0 {
-			var res2 []keybase1.APIUserSearchResult
-			res2 = append(res2, contactsRes...)
-			res2 = append(res2, res...)
-			res = res2
+			mctx.Warning("Failed to do contacts search: %s", err)
+		} else {
+			res = append(res, contactsRes...)
+			// Sort first - we are going to be deduplicating on usernames,
```

Comment:
What happened to deduplicating on username and display label? Keep in mind that one name can show up multiple times if you have multiple forms of match.

e.g. One contact has a phone number and an email. You searched for their contact name, so each phone number should appear as an entry, as should their email.

Reply:
I don't follow, did we used to have a feature and this removes it? I may be a bit lost of what's going on in different PRs.

This one had a specific things to fix in mind - so two result from both resolved contact and keybase search do not show up for the same user.

Reply:
I'm worried that this deduplication will remove too many contacts, instead of returning multiple entries per contact as is sometimes required.